test/util/populate/walk.go

```diff
@@ -0,0 +1,140 @@
+package populate
```

Comment:
I think that the above suggestions should simplify unit test coverage to the point where it's not necessary to add this file, and I would prefer that.